crates/anvil/tests/it/utils.rs

```diff
@@ -21,6 +28,45 @@ pub fn contract_addresses(chain: Chain) -> Vec<Address> {
     ]
 }
 
+pub fn http_provider(http_endpoint: &str) -> AlloyRetryProvider {
+    get_http_provider(http_endpoint)
+}
+
+pub fn http_provider_with_signer(
+    http_endpoint: &str,
+    signer: EthereumSigner,
+) -> RetryProviderWithSigner {
+    AlloyProviderBuilder::new(http_endpoint)
+        .build_with_signer(signer)
+        .expect("failed to build Alloy HTTP provider with signer")
+}
+
+pub fn ws_provider(ws_endpoint: &str) -> AlloyRetryProvider {
+    AlloyProviderBuilder::new(ws_endpoint).build().expect("failed to build Alloy WS provider")
+}
+
+pub fn ws_provider_with_signer(
+    ws_endpoint: &str,
+    signer: EthereumSigner,
+) -> RetryProviderWithSigner {
+    AlloyProviderBuilder::new(ws_endpoint)
+        .build_with_signer(signer)
+        .expect("failed to build Alloy WS provider with signer")
+}
+
+pub async fn ipc_provider(ipc_endpoint: &str) -> AlloyRetryProvider {
+    AlloyProviderBuilder::new(ipc_endpoint).build().expect("failed to build Alloy IPC provider")
+}
+
+pub async fn ipc_provider_with_signer(
+    ipc_endpoint: &str,
+    signer: EthereumSigner,
+) -> RetryProviderWithSigner {
+    AlloyProviderBuilder::new(ipc_endpoint)
+        .build_with_signer(signer)
+        .expect("failed to build Alloy IPC provider with signer")
+}
+
```

Comment:
We probably will want to include the recommended layers in `try_get_http_provider` at some point